salt/metalk8s/volumes/cleanup.sls

```diff
@@ -0,0 +1,56 @@
+# Clean already attached loop devices before applying highstate, to avoid
+# creating duplicates for a same sparse file.
+# This state can cleanup either "manually", for upgrade, or by disabling the
+# systemd units, for downgrade.
+{%- set mode = pillar.get('cleanup_mode', 'upgrade') %}
+
+include:
+  - .installed
+
+{%- if '_errors' in pillar.metalk8s.volumes %}
+
+Cannot proceed with volume cleanup (pillar errors):
+  test.fail_without_changes:
+    - comment: 'Errors: {{ pillar.metalk8s.volumes._errors | join (", ") }}'
+
+{%- else %}
+  {%- set sparse_volumes = pillar.metalk8s.volumes.values()
+                         | selectattr('spec.sparseLoopDevice', 'defined')
+                         | list %}
+
+  {%- if not sparse_volumes %}
+
+Nothing to cleanup:
+  test.succeed_without_changes: []
+
+  {%- else %}
+    {%- for volume in sparse_volumes %}
+      {%- set volume_name = volume.metadata.name %}
+      {%- set volume_id = volume.metadata.uid %}
+      {%- if mode == 'upgrade' %}
```

Comment:
Wow ! I do not really like this "mode" stuff, I would prefer to check the node version (which is the desired version) to know if we go to a 2.6.x or 2.7.x

Reply:
OK let me change that :+1: